contracts/DelFiPrice.sol

```diff
@@ -9,40 +9,60 @@ import "./OpenOracleView.sol";
-     */
+    /// @notice The event emitted when a price is written to storage
     event Price(string symbol, uint64 price);
 
-    /**
-     * @notice The mapping of medianized prices per symbol
-     */
+    /// @notice The reporter address whose prices checked against the median for safety
+    address anchor;
+
+    /// @notice The highest ratio of the new median price to the anchor price that will still trigger the median price to be updated
+    uint256 upperBoundAnchorRatio;
+
+    /// @notice The lowest ratio of the new median price to the anchor price that will still trigger the median price to be updated
+    uint256 lowerBoundAnchorRatio;
+
+    /// @notice The mapping of medianized prices per symbol
     mapping(string => uint64) public prices;
 
-    constructor(OpenOraclePriceData data_, address[] memory sources_) public OpenOracleView(data_, sources_) {}
+    /**
+     * @param data_ Address of the Oracle Data contract
+     * @param sources_ The reporter addresses whose prices will be used to calculate the median
+     * @param anchor_ The reporter address whose prices checked against the median for safety
+     * @param anchorToleranceMantissa_ The tolerance allowed between the anchor and median. A tolerance of 1e17 means a new median that is 10% off from the anchor will still be saved
+     */
+    constructor(OpenOraclePriceData data_, address[] memory sources_, address anchor_, uint anchorToleranceMantissa_) public OpenOracleView(data_, sources_) {
+        anchor = anchor_;
+        require(anchorToleranceMantissa_ < 1e18, "Anchor Tolerance is too high");
+        upperBoundAnchorRatio = 1e18 + anchorToleranceMantissa_;
+        lowerBoundAnchorRatio = 1e18 - anchorToleranceMantissa_;
+    }
 
     /**
      * @notice Primary entry point to post and recalculate prices
-     * @dev We let anyone pay to post anything, but only sources count for prices.
+     * @dev We let anyone pay to post anything, but only sources count for prices
      * @param messages The messages to post to the oracle
      * @param signatures The signatures for the corresponding messages
      */
     function postPrices(bytes[] calldata messages, bytes[] calldata signatures, string[] calldata symbols) external {
         require(messages.length == signatures.length, "messages and signatures must be 1:1");
 
-        // Post the messages, whatever they are
+        // Post the messages
         for (uint i = 0; i < messages.length; i++) {
             OpenOraclePriceData(address(data)).put(messages[i], signatures[i]);
         }
 
         // Recalculate the asset prices for the symbols to update
         for (uint i = 0; i < symbols.length; i++) {
             string memory symbol = symbols[i];
+            uint64 medianPrice = medianPrice(symbol, sources);
+            uint64 anchorPrice = OpenOraclePriceData(address(data)).getPrice(anchor, symbol);
+            uint256 anchorRatioMantissa = uint256(medianPrice) * 1e18 / anchorPrice;
```

Comment:
Do we test/handle when anchorPrice is 0?

Reply:
good point, working on it